### PR TITLE
rewrite shuffleTablets to be clearer and more efficient

### PR DESCRIPTION
**Pull request by demmer**
## Description
Implement a simple single-pass scan through the list of tablets that both randomly shuffles the set and pulls the same-cell hosts to the front of the list.

This is the same functionality as the old code, just with a simpler algorithm that only needs to loop once through the set of tablets so it's a marginally more efficient approach.

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported to release branches
-   [ ] If this change is to be back-ported to previous releases, a justification is included in the PR description
-   [x] Tests were added or are not required
-   [ ] Did the new or modified tests pass consistently locally and on CI?
-   [x] Documentation was added or is not required

### CODE REVIEW

**Comment by deepthi** [comment]
@demmer you need to signoff the commit and force-push.

**Comment by demmer** [comment]
@deepthi Pushed again with the Signed-Off.

File: go/vt/vtgate/tabletgateway.go [hunk 1]
```diff
@@ -377,50 +377,25 @@ func (gw *TabletGateway) getStatsAggregator(target *querypb.Target) *TabletStatu
...
-		diffCell = gw.nextTablet(cell, tablets, diffCell, length, false)
-		// either no more diffs or no more same cells should stop the iteration
-		if sameCell < 0 || diffCell < 0 {
-			break
-		}
 
-		if sameCell < diffCell {
-			// fast forward the `sameCell` lookup to `diffCell + 1`, `diffCell` unchanged
-			sameCell = diffCell + 1
+	// Randomly shuffle the list of tablets, putting the same-cell hosts at the front
```
Comment (deepthi) [inline go/vt/vtgate/tabletgateway.go:381]: nit: I'd always use tablet/tablets instead of host/hosts. Those are the Vitess terms.

**Review by deepthi — APPROVED** [review]
Good to see you back :)

**Review by GuptaManan100 — APPROVED** [review]
Love it!

**Comment by deepthi** [comment]
Let's give @demmer a chance to address feedback and then we can merge.

File: go/vt/vtgate/tabletgateway.go (hunk 1 above)
Comment (demmer) [reply go/vt/vtgate/tabletgateway.go:381]: ok. edited.

**Comment by deepthi** [comment]
@demmer DCO failed again. I guess the latest commit hasn't been signed off properly.

**Comment by demmer** [comment]
Oops sorry -- pushed a rework

**Pull request merged**